### PR TITLE
Refactoring Result node's hash filter code.

### PR DESCRIPTION
**Pull request by hlinnaka**
See commit messages.

I had made some of these changes as part of PR #6327, but that PR is large enough anyway, so it seems better to handle these separately.

### CODE REVIEW

File: src/backend/executor/nodeResult.c [hunk 1]
```diff
@@ -260,53 +260,12 @@ TupleMatchesHashFilter(ResultState *node, TupleTableSlot *resultSlot)
 	Assert(resultNode);
 	Assert(!TupIsNull(resultSlot));
 
-	if (resultNode->hashFilter)
+	if (node->hashFilter)
```
Comment (melanieplageman) [inline src/backend/executor/nodeResult.c]: Why keep `resultNode`? you don't use it except asserting it isn't NULL. Is that necessary?

File: src/test/regress/sql/rangefuncs_cdb.sql [hunk 2]
```diff
@@ -477,4 +477,36 @@ DROP FUNCTION dup(anyelement);
...
+begin
+   RAISE NOTICE 'notice_srf called in segment %', gp_execution_segment();
+
+   RETURN NEXT 'foo';
+   RETURN NEXT 'bar';
+end;
+$$ LANGUAGE plpgsql IMMUTABLE;
+
+-- gpdiff suppresses identical NOTICEs coming from multiple segments. But we
+-- specifically want to check that we get the NOTICE only from one segment
```
Comment (melanieplageman) [inline src/test/regress/sql/rangefuncs_cdb.sql]: "segment segment" -> "segment"

Comment (hlinnaka) [reply src/test/regress/sql/rangefuncs_cdb.sql]: fixed, thanks

**Comment by hlinnaka** [comment]
Pushed, thanks for the review!

> somewhat not the concern of this PR but I wish RelationGetPartitioningKey was named something else--like RelationGetPolicy

Agreed. I didn't change that in this PR, since it's a separate thing, but I agree it it would be worth changing..

**Pull request merged**